brave/app/scripts/controllers/bitgo.js

```diff
@@ -0,0 +1,197 @@
+      : 'http://localhost:3000/'
+    this.store = new ObservableStore(initState)
+    this.keyringController = opts.keyringController
+    chrome.braveWallet.getProjectID((projectId) => {
+      this.braveServiceKey = projectId
+    })
+  }
+
+  request (path, opts) {
+    const url = `${this.proxyOrigin}${path}`
+    const headers = new window.Headers()
+    headers.append('x-brave-key', this.braveServiceKey)
+    const req = new window.Request(url, Object.assign({ headers }, opts))
+    return window.fetch(req)
+  }
+
+  async unlockAndSetKey (password) {
+    this.password = password
+    await this.keyringController.submitPassword(this.password)
+    this.encryptionKey = await this.keyringController._getSubkey('ethwallet-encryptor')
+  }
+
+  async createWallet (coin) {
+    if (!supportedCoins[coin]) {
+      throw new Error(`Coin type ${coin} is not supported.`)
+    }
+    const seed = await promiseGetSeed(uint8ToArrayBuf(this.encryptionKey))
+    const userKeychain = bgUtil.createKeychain(coin, seed)
+    const backupKeychain = bgUtil.createKeychain(coin, seed)
+    const req = {
+      coin,
+      userPub: userKeychain.pub,
+      backupPub: backupKeychain.pub
+    }
+    console.log({req})
+    const response = await this.request('create-wallet', {
+      method: 'POST',
+      body: JSON.stringify(req)
+    })
+    console.log({response})
+    if (!response.ok) {
+      return
+    }
+    const resp = await response.json()
+    if (!resp.error) {
+      const state = this.store().getState()
+      const bitgoWallets = state.bitgoWallets || {}
+      bitgoWallets[resp.id] = Object.assign(req, resp)
+      this.store.updateState({
+        bitgoWallets
+      })
+    }
+    return resp
+  }
+
+  async getBalance (id) {
+    const bitgoWallets = this.store().getState().bitgoWallets
+    const bitgoWallet = bitgoWallets[id]
+    if (!bitgoWallet) {
+      return
+    }
+    const response = await this.request('get-balance', {
+      method: 'GET',
+      body: JSON.stringify({
+        id,
+        coin: bitgoWallet.coin
+      })
+    })
+    if (!response.ok) {
+      return
+    }
+    const resp = await response.json()
+    if (!resp.error) {
+      Object.assign(bitgoWallet, resp)
+      this.store.updateState({
+        bitgoWallets
+      })
+    }
+    return resp
+  }
+
+  async getTransfers (id) {
+    const bitgoWallets = this.store().getState().bitgoWallets
+    const bitgoWallet = bitgoWallets[id]
+    if (!bitgoWallet) {
+      return
+    }
+    const response = await this.request('get-transaction-history', {
+      method: 'GET',
+      body: JSON.stringify({
+        id,
+        coin: bitgoWallet.coin
+      })
+    })
+    if (!response.ok) {
+      return
+    }
+    const resp = await response.json()
+    if (!resp.error) {
+      bitgoWallet.transfers = response.transfers
+      this.store.updateState({
+        bitgoWallets
+      })
+    }
+    return resp
+  }
+
+  async sendTx (id, amount, destinationAddress) {
```

Comment:
just an FYI, `amount` is the amount in base value (ex: satoshis) as a string not a number